src/replica/HttpDirectorIndexModule.cc

```diff
@@ -78,123 +78,114 @@ json HttpDirectorIndexModule::_buildDirectorIndex() {
     debug(__func__, "rebuild=" + bool2str(rebuild));
 
     auto const database = config->databaseInfo(databaseName);
-    if (database.isPublished and not allowForPublished) {
+    if (database.isPublished and !allowForPublished) {
         throw HttpError(__func__, "database '" + database.name +
                                           "' is already published. Use 'allow_for_published' option to "
                                           "override the restriction.");
     }
-    vector<string> const directorTables =
-            directorTableName.empty() ? database.directorTables() : vector<string>({directorTableName});
-
-    // Pre-screen parameters of the table(s).
-    map<string, string> primaryKeyColumn;
-    map<string, string> primaryKeyColumnType;
-    map<string, string> chunkIdColNameType;
-    map<string, string> subChunkIdColNameType;
-
-    for (auto&& tableName : directorTables) {
-        auto const table = database.findTable(tableName);
-        if (!table.isDirector) {
-            string const msg = "table '" + table.name +
-                               "' is not configured as the director table in database '" + database.name +
-                               "'";
-            throw HttpError(__func__, msg);
-        }
-        if (table.directorTable.primaryKeyColumn().empty()) {
-            string const msg =
-                    "director table has not been properly configured in database '" + database.name + "'";
-            throw HttpError(__func__, msg);
-        }
-        primaryKeyColumn[table.name] = table.directorTable.primaryKeyColumn();
+    auto const table = database.findTable(directorTableName);
+    if (!table.isDirector) {
+        string const msg = "table '" + table.name +
+                           "' is not configured as the director table in database '" + database.name + "'";
```

Comment:
Should be "... not configured as a director table ..." (there can be more than one director in the database)

Reply:
Good point!